src/Microsoft.Health.Fhir.Core/Configs/ExportJobFormatConfiguration.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.Core.Configs
+{
+    public class ExportJobFormatConfiguration
+    {
+        /// <summary>
+        /// The name of the format. This is how the format is referenced in the _format parameter when creating a new export job.
+        /// </summary>
+        public string Name { get; set; } = string.Empty;
+
+        /// <summary>
+        /// The format definition string. An export job's format is used to create the folder stucture inside the container.
+        /// The format is defined by tags and characters. Supported tags are defined below. The / character is used to indicate a subfolder.
+        /// <timestamp> - Places a timestamp corisponding to the time the export job was enqueued.
+        /// <resourcename> - The name of the resource currently being exported.
+        /// <id> - The GUID id of the export job.
+        /// </summary>
+        public string Format { get; set; }
+
+        /// <summary>
+        /// Whether the format is the default format for when no container is specified by the user.
+        /// </summary>
+        public bool NoContainerDefault { get; set; } = false;
+
+        /// <summary>
+        /// Whether the format is the default format for when a container is specified by the user.
+        /// </summary>
+        public bool ContainerDefault { get; set; } = false;
```

Comment:
From how it's used, it feels more like a bool switch to determine is this container specific or not.
Any chance this can be reduced to just one variable? (Container/NotContainer)
Or even moved it out, and having two configurations one container specific and other non container?

I understand desire to have default, but nothing stops people to define multiple default configurations since there is no validation, so maybe just document what first configuration is default one if no name specified?

Reply:
Ok, maybe two collections for container specified/not specified is a bit of overkill, it doesn't make sense from user perspective to have to different collections.

I would probably go following route:
If user didn't specify format, just don't pick configuration and leave old code in place.
If user specify format, and it's not found, I would fail with error to user.
If user specify format and it's found, use it despite is it container value.

That would allow you to preserve old behavior, and would remove this two flags, which from user standpoint only bring confusion. 

---
In reply to: [518329969](https://github.com/microsoft/fhir-server/pull/1422#discussion_r518329969) [](ancestors = 518329969)

Reply:
That been said I actually start questioning why we even have this collection in first place?

Why user can't just pass us `Format` string?

User need to learn how `Format` works anyway and in current moment he need to do roundtrip of going to app settings and looking for what was the format name for this specific format.

---
In reply to: [518341229](https://github.com/microsoft/fhir-server/pull/1422#discussion_r518341229) [](ancestors = 518341229,518329969)

Reply:
I agree that if a user enters a name that isn't found an error should be returned, I'll add that. The thought behind having two defaults is because currently we have two different behaviors depending on if a container is specified or not. The reason I added both was to minimize the hardcoded values we would need. 

As for just passing in the format, I wanted the users to have the ability to set up a standard that they could specify. That way not all users needed to know about the export format. The server admins would set up any that the organization needed and then the format could just be referenced by name. I'm open to just having them pass in the format. @CaitlinV39 thoughts?

Reply:
@LTA-Thinking  and @Ivanidzo4ka unless its too inefficient, I like that the format could be setup and then other users would be able to just leverage it without needing to know the full format.

Reply:
Fine by me. My main concern is that two extra properties in configuration.
So here is my reasoning why I would prefer to hardcode them in code rather than expose as part of configuration.

I don't think there is much value in them other than back compatibility.  User already specifying name of format, it can be confusing if his format would change to something else depending on passed container value.
It's hard to server admin to configure, because this two new configuration bring more options, and they confusing.
Why presence of container is important? It used to be, but with passing format name, should it still be important?
I think it's better to have some hardcoded code for back-compat in code base, than force users to scratch their heads in order to figure out what they should put in this fields.

Just in case, I'm not saying there is no need to keep old code, I'm in favor in full export configuration change, just in place where we choose configuration we can check is format passed or not, and pass configuration 
format based on container presence if there is no format.

---
In reply to: [518920366](https://github.com/microsoft/fhir-server/pull/1422#discussion_r518920366) [](ancestors = 518920366)

Reply:
I removed the two tags and replaced them with a single "Default" property. Neither of the included formats have the property set. If there is no default format and no format is entered the behavior will be identical to current behavior.